feat: add Box component (#551)
medly/medly-components: packages/core/src/components/Box/Box.stories.mdx

import { Box } from './Box.tsx';
import { fontVariants, shadows } from './Box.stories';
import { withKnobs, color, select, boolean, number } from '@storybook/addon-knobs';
import { Preview, Meta, Props, Story } from '@storybook/addon-docs/blocks';
import { defaultTheme } from '@medly-components/theme';

<Meta title="Core" component={Box} />

# Box

A simple box component with customizable properties.

> You can play with the props in the **Canvas** tab.

Reviewer: This line can be removed since user will already know their way around the storybook
```suggestion
```

Author: That's assuming the user already know how to use Canvas tab but incase someone is new then it will be helpful.
Also same is there for every core component in such as [Button Component](https://www.medlycomponents.com/?path=/docs/core--button)

> 

Reviewer: ```suggestion
```

<Preview withToolbar>
    <Story name="Box" decorators={[withKnobs]}>
        <Box
            fontVariant={select('Font Variant', fontVariants, 'h2')}
            color={color('Text Color', 'white')}

>             bg={color('Background Color', 'black')}

Reviewer: We can use Medly colors to match them more to the theme of the documentation

            isLoading={boolean('Is Loading', false)}
            px={number('Padding x-axis', 4)}
            py={number('Padding y-axis', 4)}
            mx={number('Margin x-axis', 2)}
            my={number('Margin y-axis', 2)}
            borderRadius={number('Border radius', 2)}
            borderColor={color('Border color', 'tomato')}
            borderWidth={number('Border width', 1)}
            shadow={select('Shadow', shadows, 'M')}
            shadowColor={color('Shadow Color', 'rgba(0, 0, 0, 0.2)')}
        >
            Box Text
        </Box>
    </Story>
</Preview>

### Padding and Margin

Padding can be defined with `px` and `py`.

> -   `px` means padding along x-axis.

Reviewer: ```suggestion
-   `px` - padding along x-axis.
```

> -   `py` means padding along y-axis.

Reviewer: ```suggestion
-   `py` - padding along y-axis.
```

Margin can be defined with `mx` and `my`.

> -   `mx` means margin along x-axis.

Reviewer: ```suggestion
-   `mx` - margin along x-axis.
```

> -   `my` means margin along y-axis.

Reviewer: ```suggestion
-   `my` - margin along y-axis.
```

> **Note**: Values for padding and margin are defined as a `number` and then converted into `rem` using `(value * 0.25rem)`

<Preview>
    <Box px={4} py={4} mx={0} my={0} borderWidth={1} borderColor="tomato">
        Box with padding
    </Box>
    <Box mx={8} my={4} px={0} py={0} borderWidth={1} borderColor="tomato">
        Box with margin
    </Box>
</Preview>

### Color and Background Color

Color and Background color can be provided to box component as `color` and `bg` respectively.

<Preview>
    <Box color="white" bg="black" mx={2}>
        Box with white color and black background
    </Box>
    <Box color="white" bg={defaultTheme.colors.blue[500]} mx={2}>
        Box with white color and blue background
    </Box>
</Preview>

> ### Height and width

Reviewer: ```suggestion
### Height and Width
```

> Height and widht can be defined as any css unit, eg: `100%`, `10rem`, `200px`

Reviewer: ```suggestion
_height_ and _width_ can be defined as any CSS unit ( rem | px | percent | ... ). 
```
